processor/src/hasher/tests.rs

```diff
@@ -209,27 +247,70 @@ fn hasher_update_merkle_root() {
     check_merkle_path(&trace, 48, leaves[6], &path6, 6, MR_UPDATE_OLD);
     check_merkle_path(&trace, 72, new_leaf6, &path6, 6, MR_UPDATE_NEW);
     check_merkle_path(&trace, 96, new_leaf3, &path3_2, 3, MR_UPDATE_OLD);
     check_merkle_path(&trace, 120, new_leaf3_2, &path3_2, 3, MR_UPDATE_NEW);
+
+    // make sure sibling table hints were built correctly
+    let expected_sibling_hints = vec![
+        // first update
+        (0, SiblingTableUpdate::SiblingAdded(0)),
+        (8, SiblingTableUpdate::SiblingAdded(1)),
+        (16, SiblingTableUpdate::SiblingAdded(2)),
+        (24, SiblingTableUpdate::SiblingRemoved(0)),
+        (32, SiblingTableUpdate::SiblingRemoved(1)),
+        (40, SiblingTableUpdate::SiblingRemoved(2)),
+        // second update
+        (48, SiblingTableUpdate::SiblingAdded(3)),
+        (56, SiblingTableUpdate::SiblingAdded(4)),
+        (64, SiblingTableUpdate::SiblingAdded(5)),
+        (72, SiblingTableUpdate::SiblingRemoved(3)),
+        (80, SiblingTableUpdate::SiblingRemoved(4)),
+        (88, SiblingTableUpdate::SiblingRemoved(5)),
+        // third update
+        (96, SiblingTableUpdate::SiblingAdded(6)),
+        (104, SiblingTableUpdate::SiblingAdded(7)),
+        (112, SiblingTableUpdate::SiblingAdded(8)),
+        (120, SiblingTableUpdate::SiblingRemoved(6)),
+        (128, SiblingTableUpdate::SiblingRemoved(7)),
+        (136, SiblingTableUpdate::SiblingRemoved(8)),
+    ];
+    assert_eq!(expected_sibling_hints, aux_hints.sibling_hints);
+
+    let expected_sibling_rows = vec![
+        // first update
+        SiblingTableRow::new(Felt::new(3), path3[0]),
+        SiblingTableRow::new(Felt::new(3 >> 1), path3[1]),
+        SiblingTableRow::new(Felt::new(3 >> 2), path3[2]),
+        // second update
+        SiblingTableRow::new(Felt::new(6), path6[0]),
+        SiblingTableRow::new(Felt::new(6 >> 1), path6[1]),
+        SiblingTableRow::new(Felt::new(6 >> 2), path6[2]),
+        // third update
+        SiblingTableRow::new(Felt::new(3), path3_2[0]),
+        SiblingTableRow::new(Felt::new(3 >> 1), path3_2[1]),
+        SiblingTableRow::new(Felt::new(3 >> 2), path3_2[2]),
+    ];
+    assert_eq!(expected_sibling_rows, aux_hints.sibling_rows);
 }
 
 // HELPER FUNCTIONS
 // ================================================================================================
 
 /// Builds an execution trace for the provided hasher. The trace must have the number of rows
 /// specified by num_rows.
-fn build_trace(hasher: Hasher, num_rows: usize) -> Vec<Vec<Felt>> {
+fn build_trace(hasher: Hasher, num_rows: usize) -> (Vec<Vec<Felt>>, AuxTraceBuilder) {
+    let aux_hints = hasher.aux_trace.clone();
     let mut trace = (0..TRACE_WIDTH)
         .map(|_| vec![Felt::new(0); num_rows])
         .collect::<Vec<_>>();
     let mut fragment = TraceFragment::trace_to_fragment(&mut trace);
     hasher.fill_trace(&mut fragment);
-    trace
+    (trace, aux_hints)
```

Comment:
Since an `AuxTraceBuilder` is returned from the `fill_trace` function, do we actually need this `clone`?  Also, a nit: it might be clearer to rename `aux_hints` since we're no longer referring to it that way.

Reply:
Good catch! Fixed.